Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -36,10 +36,21 @@ public object Create(object request, ISpecimenContext context)
                 return new NoSpecimen(request);
             }
 
-            var name = Guid.NewGuid().ToString();
-            var host = fictitiousDomains[(uint)name.GetHashCode() % 3];
+            if (context == null)
+            {
+                return new NoSpecimen(request);

```

Comment:
That makes total sense and if I had looked more closely, `UriGenerator` adheres to that policy as well of throwing `ArgumentNullException` when the context is null since it requires the context, but other generators don't evaluate it at all if they don't make use of it.   The build script caught this when I modified `MailAddressGenerator` and I missed that connection between the precondition and the exception.